tests/test_buffer.py

```diff
@@ -62,3 +62,42 @@ def test_sendbuf_no_buffer(allocator, contiguous_array):
 def test_recvbuf_no_buffer(allocator, contiguous_array):
     with recv_buffer(allocator, contiguous_array) as recvbuf:
         assert recvbuf is contiguous_array
+
+
+def test_buffer_cache_appends(allocator, backend):
+    """Test buffer with the same key are appended while not in use for potential reuse"""
+    if backend == "gt4py_cupy":
+        pytest.skip("gt4py gpu backend cannot produce contiguous arrays")
+    BUFFER_CACHE.clear()
+    assert len(BUFFER_CACHE) == 0
+    shape = (10, 10, 10)
+    b0_0 = Buffer.pop_from_cache(allocator, shape, float)
+    b0_0.array.fill(42)
+    assert len(BUFFER_CACHE) == 1
+    b0_1 = Buffer.pop_from_cache(allocator, shape, float)
+    b0_1.array.fill(23)
+    assert b0_0._key == b0_1._key
+    assert (b0_0.array != b0_1.array).all()
+    assert len(BUFFER_CACHE) == 1
+    assert len(next(iter(BUFFER_CACHE.items()))[1]) == 0
+    Buffer.push_to_cache(b0_0)
+    Buffer.push_to_cache(b0_1)
+    assert len(next(iter(BUFFER_CACHE.items()))[1]) == 2
```

Comment:
I'm thrown off by the fact that there's a [1] index here. Does BUFFER_CACHE have two items in it for some reason? Shouldn't it have only one key since both buffers have the same key?

Reply:
I reworked the test to be more readable, including using keys and comments. It's quite confusing because the cache line is an array. Using the `key` we read better what I am accessing